pkg/build/netbsd.go

```diff
@@ -66,3 +73,51 @@ func (ctx netbsd) clean(kernelDir string) error {
 	// machine.
 	return nil
 }
+
+// Copy the compiled kernel to the qemu disk image using ssh
+func CopyKernelToDisk(outputDir string) error {
+	temp := []byte(`
+{
+	"snapshot": false,
+	"mem": 1024
+}	`)
+	VMconfig := (*json.RawMessage)(&temp)
+	// Create config for booting the disk image
+	cfg := &mgrconfig.Config{
+		Workdir:      outputDir,
+		Image:        filepath.Join(outputDir, "image"),
+		SSHKey:       filepath.Join(outputDir, "key"),
+		SSHUser:      "root",
+		TargetOS:     "netbsd",
+		TargetArch:   "amd64",
+		TargetVMArch: "amd64",
+		Type:         "qemu",
+		VM:           *VMconfig,
+	}
+	// Create a VM pool
+	pool, err := vm.Create(cfg, false)
+	if err != nil {
+		return fmt.Errorf("failed to create a VM Pool : %v", err)
+	}
+	// Create a VM instance (We need only one)
+	inst, err := pool.Create(0)
+	if err != nil {
+		return fmt.Errorf("failed to create the VM Instance : %v", err)
+	}
+	defer inst.Close()
+	// Copy the kernel into the disk image and replace it
+	// This makes use of the fact that the file is copied by default to /
+	kernel, err := inst.Copy(filepath.Join(outputDir, "netbsd"))
+	if err != nil {
+		return fmt.Errorf("error Copying the kernel %v: %v", kernel, err)
+	}
+	// Run poweroff so that the copied image is stored properly
+	// Due to some reason poweroff command isn't recognized over ssh
+	// hence we use /sbin/poweroff. 
+	outc, errc, err := inst.Run(time.Minute, nil, "/sbin/poweroff")
```

Comment:
This shutdown and sleep does not look like a good solution. It still don't know if it succeeded or not, maybe it failed during the sleep. And these outc and errc printed in the error below are not actually output and error, these are _channels_ though which you can receive output and error.

What have you tried with MonitorExecution and how did not it work?
You can see usage examples in tools/syz-runtest/runtest.go and pkg/instance/instance.go (the ones that use vm.ExitNormal -- which means the monitored program can only exit normally).

Reply:
I have added a patch with the vm.MonitorExecution. The issue that I am facing with it is that the report is always that it "lost connection with the machine" (Basically since we are powering it off). The code however still copies the kernel properly to the disk image.
Also there is a matter of how and what I would return after executing the function.

Reply:
Well, "lost connection with the machine" is expected if we doing poweroff. But can we get back to running "sync"?

Reply:
sync(8) is a command that shall be called, not necessarily needed but better to be safe.

Reply:
Okay fixed!

Reply:
I don't see what has changed.

Reply:
And there are few other open comments.

Reply:
Sorry! I just pushed the changes